api/src/test/java/org/openmrs/module/fhir2/api/search/ObservationSearchQueryTest.java

```diff
@@ -1585,6 +1585,112 @@ public void searchForLastnObs_shouldHandleNoCodeRequest() {
 		assertThat(resultList, everyItem(anyOf(allOf(is(instanceOf(Observation.class))))));
 	}
 	
+	@Test
+	public void searchForLastnEncountersObs_shouldHandleNormalRequest() {
+		ReferenceAndListParam referenceParam = new ReferenceAndListParam();
+		ReferenceParam patient = new ReferenceParam();
+		
+		patient.setValue(PATIENT_UUID);
+		
+		referenceParam.addValue(new ReferenceOrListParam().add(patient));
+		
+		TokenAndListParam categories = new TokenAndListParam().addAnd(new TokenParam().setValue("laboratory"));
+		
+		TokenAndListParam code = new TokenAndListParam().addAnd(
+		    new TokenParam().setSystem(FhirTestConstants.LOINC_SYSTEM_URL).setValue(LOINC_SYSTOLIC_BP),
+		    new TokenParam().setSystem(FhirTestConstants.CIEL_SYSTEM_URN).setValue(OBS_CONCEPT_ID));
+		
+		SearchParameterMap theParams = new SearchParameterMap().addParameter(FhirConstants.CODED_SEARCH_HANDLER, code)
+		        .addParameter(FhirConstants.CATEGORY_SEARCH_HANDLER, categories)
+		        .addParameter(FhirConstants.MAX_SEARCH_HANDLER, new NumberParam(2))
+		        .addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER, referenceParam)
+		        .addParameter(FhirConstants.LASTN_ENCOUNTERS_SEARCH_HANDLER, new StringParam());
+		
+		IBundleProvider results = search(theParams);
+		
+		assertThat(results, notNullValue());
+		List<IBaseResource> resultList = get(results);
+		
+		assertThat(resultList.size(), equalTo(2));
+		assertThat(resultList, everyItem(anyOf(allOf(is(instanceOf(Observation.class))))));
+		
```

Comment:
In the tests for search query, it might be worthwhile to check the number of distinct Encounter uuids related to the output Observations.
`observation.getEncounter().getReferenceElement().getIdPart()`

Reply:
Yes, I'll add those assertions.

Reply:
Should we check distinct Encounters or rather extract the `encounterDatetime` and check the count of distinct `encounterDatetime`?

Reply:
Yeah, we can do that as well.